src/forwarders.interfaces.ts

```diff
@@ -1,29 +1,37 @@
 import { SDKEvent, SDKEventCustomFlags } from './sdkRuntimeModels';
 import { Dictionary } from './utils';
-import { IKitConfigs } from './configAPIClient';
+import { IKitConfigs, IKitFilterSettings } from './configAPIClient';
 import { IdentityApiData } from '@mparticle/web-sdk';
 import {
     IMParticleUser,
     ISDKUserIdentity,
     UserAttributes,
 } from './identity-user-interfaces';
 
-// TODO: https://go.mparticle.com/work/SQDSDKS-6035
-export type Kit = Dictionary;
+// TODO: https://go.mparticle.com/work/SQDSDKS-4475
 export type MPForwarder = Dictionary;
 
 // The state of the kit when accessed via window.KitName via CDN
 // or imported as an NPM package
 export interface UnregisteredKit {
-    register(config): void;
+    constructor: () => void;
+    register(config: KitRegistrationConfig): void;
+    name: string;
+    suffix?: string;
 }
 
 // The state of the kit after being added to forwarderConstructors in the CDN
 // or after registered to SDKConfig.kits via NPM
 export interface RegisteredKit {
-    constructor(): void;
-    name: string;
-    getId(): number;
+    constructor: () => void;
+
+    // Applies to sideloaded kits only
+    filters: IKitFilterSettings;
```

Comment:
if it applies to sideloaded kits only, perhaps this should be optional?

Reply:
Perhaps. The issue is that in the code for `processSideloadedKits`, it treats `filters` as required. IMO we may want to change the behavior of `processSideloadedKits` to internally use a SideloadedKit interface rather than RegisteredKit. I can look into that as part of https://go.mparticle.com/work/SQDSDKS-4475